elfo-core/src/errors.rs

```diff
@@ -1,14 +1,36 @@
 use derive_more::{Display, Error};
 
+use crate::messages::{ConfigRejected, EntrypointFailed};
+
+#[derive(Debug, Display, Error)]
+#[non_exhaustive]
+#[display(fmt = "failed to start group {group} because {reason}")]
+pub struct StartError {
+    pub group: String,
+    pub reason: StartErrorReason,
+}
+
 #[derive(Debug, Display, Error)]
 #[non_exhaustive]
-pub enum StartError {
-    /// Configs must be valid at the start-up.
-    #[display(fmt = "invalid config")]
-    InvalidConfig,
+pub enum StartErrorReason {
+    #[display(fmt = "entrypoint failed: {}", _0)]
+    EntrypointFailed(#[error(not(source))] String),
+    #[display(fmt = "other: {}", _0)]
     Other(#[error(not(source))] String),
 }
 
+impl From<ConfigRejected> for StartErrorReason {
```

Comment:
This is a bit awkward. Initially, I wanted to just add `#[derive(Display)]` to the `ConfigRejected` and call `to_string()` in order to extract the failure reason. However, this produces conflicting implementations of `From` trait, since we already have this in the codebase:

```rust
impl<R: Display> From<R> for ConfigRejected {
    fn from(reason: R) -> Self {
        Self {
            reason: reason.to_string(),
        }
    }
}
```

And `ConfigRejected` having `Display` trait implemented causes this to conflict with `impl From<T> for T` from Rust itself. Replacing `Display` with `ToString` above does not solve the problem, since `ToString` is implemented for `Display`.

Without negative trait bounds, there is not much of a choice left. Let me know if you have a workaround in mind.